R/internal.R

```diff
@@ -33,7 +33,41 @@ coalesceifnull <- function(x, y) {
 #' @importFrom reticulate import
 boto <- reticulate::import("boto", delay_load = TRUE)
 
+read.text = function(pathname) {
+  if (file.exists(pathname)) {
+	      return (paste(readLines(pathname), collapse="\n"))
+  } 
+}
+
+check_aws_credentials <- function() {
+  aws_credentials_path = "~/.aws/credentials"
+  aws_configure_path = "~/.aws/configure"
+  renviron_path = "/etc/R/Renviron.site"
+  rprofile_path = ".Rprofile"
+  aws_key_env_set = Sys.getenv("AWS_ACCESS_KEY_ID") != ""
+  aws_configure_exists = file.exists(aws_configure_path) 
+  aws_credentials_exists = file.exists(aws_credentials_path)
+  # Does not need checking as both would set env var, keeping it here for completness
+  renviron_exists = file.exists(renviron_path)
+  rprofile_exists = file.exists(rprofile_path)
+  if (aws_key_env_set) {
+    log_if_verbose("AWS_ACCESS_KEY_ID is set, it will override other aws credentials") 	
+  } else if (aws_credentials_exists) {
+    if (grepl("aws_access_key_id", read.text(aws_credentials_path))) {
+      log_if_verbose("aws_access_key_id is set in ~/.aws/credentials, it will overrride other aws credentials")
+    }
+  } else if (aws_configure_exists) {
+    if (grepl("aws_access_key_id", read.text(aws_configure_path))) {
+	log_if_verbose("aws_access_key_id is set in ~/.aws/configure, it will overrride other aws credentials")
```

Comment:
how do we run in verbose?

Reply:
great question, just figured it out two days ago -> `options(redshiftTools.verbose=TRUE)`

Reply:
Same thing with zapieR and I think Payne and Parquetr.